Transform.cs

```diff
@@ -30,6 +30,15 @@ public Displacement DisplacementTo(Position target)
         internal static Position _FromVec(Vector position)
             => new Position { x = position.x, y = position.z, up = position.y };
 
+        public static Position Lerp(Position a, Position b, float t)
```

Comment:
I added this because I didn't understand how `AdvanceToTarget()`  works. can you please help me out? do I need this?

Reply:
It's basically Lerp but with an extra step of making sure that the difference between the final value and `a` is within `rangeOfChange`. Therefore, having another method `Lerp` should be better if clamping is unnecessary.